viewer/src/tools/AxisView/AxisViewTool.ts

```diff
@@ -0,0 +1,321 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+
+import * as THREE from 'three';
+import ComboControls from '@cognite/three-combo-controls';
+import TWEEN from '@tweenjs/tween.js';
+import _ from 'lodash';
+
+import { Cognite3DViewer } from '../../migration';
+import { Cognite3DViewerToolBase } from '../Cognite3DViewerToolBase';
+import { AxisBoxConfig, defaultAxisBoxConfig, AxisBoxFaceConfig, Corner, Absolute, Relative } from './types';
+
+export class AxisViewTool extends Cognite3DViewerToolBase {
+  private readonly _layoutConfig: AxisBoxConfig;
+
+  private readonly _boxFaceGeometry: THREE.PlaneGeometry;
+
+  private readonly _viewer: Cognite3DViewer;
+
+  private readonly _interactiveObjects: THREE.Mesh[];
+  private readonly _raycastCamera: THREE.OrthographicCamera;
+  private readonly _raycaster: THREE.Raycaster;
+
+  private _screenPosition: THREE.Vector2;
+
+  private dynamicUpdatePosition = () => {};
+
+  constructor(viewer: Cognite3DViewer, config?: AxisBoxConfig) {
+    super();
+
+    this._screenPosition = new THREE.Vector2();
+    this._boxFaceGeometry = new THREE.PlaneGeometry(0.85, 0.85, 1, 1);
+
+    this._raycastCamera = new THREE.OrthographicCamera(-1, 1, 1, -1, -1, 1);
+    this._raycaster = new THREE.Raycaster();
+
+    this._viewer = viewer;
+
+    this._layoutConfig = _.merge(_.cloneDeep(defaultAxisBoxConfig), config);
```

Comment:
```suggestion
    this._layoutConfig = _.{ ...defaultAxisBoxConfig, ...config };
```
will *almost* do the same, but won't create a deep clone, but do you really need that here?